packages/reaction-core/client/subscriptions.js

```diff
@@ -3,36 +3,44 @@
  * Create persistent sessions for users
  * The server returns only one record, so findOne will return that record
  * Stores into client session all data contained in server session
- * supports reactivity when server changes the serverSession
+ * supports reactivity when server changes `newSession`
  * Stores the server session id into local storage / cookies
+ *
+ * Also localStorage session could be set from the client-side. This could
+ * happen when user flush browser's cache, for example.
+ * @see https://github.com/reactioncommerce/reaction/issues/609#issuecomment-166389252
  */
-let currentSession;
-let serverSession = Random.id();
-
 Tracker.autorun(function () {
-  currentSession = Session.get("sessionId") || amplify.store(
-    "ReactionCore.session");
-  if (!currentSession) {
-    amplify.store("ReactionCore.session", serverSession);
-    Session.set("sessionId", serverSession);
-    currentSession = serverSession;
+  // we are trying to track both amplify and Session.get here, but the problem
+  // is - we can't track amplify. It just not tracked. So, to track amplify we
+  // are using dirty hack inside Accounts.loginWithAnonymous method.
+  const sessionId = amplify.store("ReactionCore.session");
+  let newSession;
+  Tracker.nonreactive(() => {
+    newSession = Random.id();
+  });
+  if (typeof sessionId !== "string") {
+    amplify.store("ReactionCore.session", newSession);
+    Session.set("sessionId", newSession);
+  }
+  if (typeof Session.get("sessionId") !== "string") {
```

Comment:
Perhaps Session.equals could be used here, whenever you use Session.get you are causing the computations to rerun, so it's important to avoid excess `Session.get`.

```
If value is a scalar, then these two expressions do the same thing:

(1) Session.get("key") === value
(2) Session.equals("key", value)
... but the second one is always better. It triggers fewer invalidations (template redraws), making your program more efficient.
```